examples/jsm/loaders/MMDLoader.js

```diff
@@ -2064,4 +2078,146 @@ class CubicBezierInterpolation extends Interpolant {
 
 }
 
+class MMDToonMaterial extends ShaderMaterial {
+
+	constructor( parameters ) {
+
+		super();
+
+		this.color = new Color( 0xffffff );
+		this.specular = new Color( 0x111111 );
+		this.shininess = 30;
+
+		this.matcap = null;
+		this.matcapCombine = null;
+
+		this.map = null;
+		this.gradientMap = null;
+
+		this.lightMap = null;
+		this.lightMapIntensity = 1.0;
+
+		this.aoMap = null;
+		this.aoMapIntensity = 1.0;
+
+		this.emissive = new Color( 0x000000 );
+		this.emissiveIntensity = 1.0;
+		this.emissiveMap = null;
+
+		this.bumpMap = null;
+		this.bumpScale = 1;
+
+		this.normalMap = null;
+		this.normalMapType = TangentSpaceNormalMap;
+		this.normalScale = new Vector2( 1, 1 );
+
+		this.displacementMap = null;
+		this.displacementScale = 1;
+		this.displacementBias = 0;
+
+		this.specularMap = null;
+
+		this.alphaMap = null;
+
+		this.envMap = null;
+		this.combine = MultiplyOperation;
+		this.reflectivity = 1;
+		this.refractionRatio = 0.98;
+
+		this.wireframeLinecap = 'round';
+		this.wireframeLinejoin = 'round';
+
+		this.flatShading = false;
+
+		parameters.defines = Object.assign( {}, MMDToonShader.defines );
+		switch ( parameters.matcapCombine ) {
+
+			case MultiplyOperation:
+				parameters.defines[ 'MATCAP_BLENDING_MULTIPLY' ] = '';
+				break;
+
+			case AddOperation:
+				parameters.defines[ 'MATCAP_BLENDING_ADD' ] = '';
+				break;
+
+		}
+
+		parameters.uniforms = UniformsUtils.clone( MMDToonShader.uniforms );
+		parameters.uniforms.diffuse.value = parameters.color;
+		parameters.uniforms.matcap.value = parameters.matcap;
+		parameters.uniforms.map.value = parameters.map;
+		parameters.uniforms.envMap.value = parameters.envMap;
+		parameters.uniforms.reflectivity.value = parameters.reflectivity;
+		parameters.uniforms.refractionRatio.value = parameters.refractionRatio;
+		parameters.uniforms.specular.value = parameters.specular;
+		parameters.uniforms.shininess.value = parameters.shininess;
+		parameters.uniforms.emissive.value = parameters.emissive;
+		parameters.uniforms.gradientMap.value = parameters.gradientMap;
```

Comment:
Should we have getters and setters like

```
get shininess() {
  return this.uniforms.shininess.value;
}

set shininess(value) {
  this.uniforms.shininess.value = value;
}

...
```

?

Otherwise users need to be aware that new MMD toon material is based on `ShaderMaterial` and they need to rewrite their code from `material.param = value` to `material.uniforms.param.value = value`. I don't think we should break the compatibility.

Reply:
Not use setter/getter on production before, let me try it.